webauthn-authenticator-rs/src/ctap2/ctap21.rs

```diff
@@ -67,240 +60,118 @@ impl<'a, T: Token, U: UiCallback> Ctap21Authenticator<'a, T, U> {
-            .get_pin_uv_auth_session(Permissions::BIO_ENROLLMENT, None, false)
-            .await?;
-
-        let mut r = self
-            .bio_with_session(
-                BioSubCommand::FingerprintEnrollBegin(timeout),
-                iface.as_ref(),
-                pin_uv_auth_token.as_ref(),
-            )
-            .await?;
-
-        trace!("began enrollment: {:?}", r);
-        let id = r.template_id.ok_or(WebauthnCError::MissingRequiredField)?;
-
-        let mut remaining_samples = r
-            .remaining_samples
-            .ok_or(WebauthnCError::MissingRequiredField)?;
-        while remaining_samples > 0 {
-            self.ui_callback
-                .fingerprint_enrollment_feedback(remaining_samples, r.last_enroll_sample_status);
-
-            r = self
-                .bio_with_session(
-                    BioSubCommand::FingerprintEnrollCaptureNextSample(id.clone(), timeout),
-                    iface.as_ref(),
-                    pin_uv_auth_token.as_ref(),
-                )
-                .await?;
-
-            remaining_samples = r
-                .remaining_samples
-                .ok_or(WebauthnCError::MissingRequiredField)?;
-        }
-
-        // Now it's enrolled, give it a name.
-        if friendly_name.is_some() {
-            self.bio_with_session(
-                BioSubCommand::FingerprintSetFriendlyName(TemplateInfo {
-                    id: id.clone(),
-                    friendly_name,
-                }),
-                iface.as_ref(),
-                pin_uv_auth_token.as_ref(),
-            )
-            .await?;
-        }
-
-        Ok(id)
+    /// [0]: https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#toggle-alwaysUv
+    pub async fn toggle_always_uv(&mut self) -> Result<(), WebauthnCError> {
+        self.config(ConfigSubCommand::ToggleAlwaysUv, true).await
     }
 
-    /// Lists all enrolled fingerprints in the device.
+    /// Sets a [minimum PIN length policy][0].
     ///
-    /// Returns [WebauthnCError::NotSupported] if the token does not support fingerprint authentication.
-    pub async fn list_fingerprints(&mut self) -> Result<Vec<TemplateInfo>, WebauthnCError> {
-        // TODO: handle CTAP_2_1_PRE version too
-        self.check_fingerprint_support().await?;
-
-        // works without authentication if alwaysUv = false?
-        let templates = self
-            .bio(BioSubCommand::FingerprintEnumerateEnrollments)
-            .await?;
-
-        Ok(templates.template_infos)
-    }
-
-    /// Renames an enrolled fingerprint.
-    pub async fn rename_fingerprint(
+    /// This is only available on authenticators which
+    /// [support configuration][Self::supports_config].
+    ///
+    /// [0]: https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#setMinPINLength
+    pub async fn set_min_pin_length(
```

Comment:
Do we actually have any devices that support this?

Reply:
The Yubikey Bio does.